packages/diagrams/frontend/sirius-components-diagrams/src/renderer/helper-lines/useHelperLines.tsx

```diff
@@ -125,44 +128,240 @@ const getHelperLines = (
+            setState({ vertical: helperLines.vertical, horizontal: helperLines.horizontal });
+            if (helperLines.snapX && change.dimensions && resizingNode.positionAbsolute) {
+              change.dimensions.width = Math.abs(resizingNode.positionAbsolute.x - helperLines.snapX);
+            }
+            if (helperLines.snapY && change.dimensions && resizingNode.positionAbsolute) {
+              change.dimensions.height = Math.abs(resizingNode.positionAbsolute.y - helperLines.snapY);
+            }
+          }
+        }
+      } else if (enabled && changes.length === 2 && changes[0] && changes[1]) {
+        const movingChange = changes[0];
+        const resizingChange = changes[1];
+        if (isMove(movingChange, false) && isResize(resizingChange)) {
+          const resizingNode = getNodes().find((node) => node.id === movingChange.id);
+          if (resizingNode) {
+            const helperLines: HelperLines = getHelperLinesForResizeAndMove(
+              resizingChange,
+              movingChange,
+              resizingNode,
+              getNodes()
+            );
+            setState({ vertical: helperLines.vertical, horizontal: helperLines.horizontal });
+            let snapOffsetX: number = 0;
+            let snapOffsetY: number = 0;
+            let parentNode = getNodes().find((node) => node.id === resizingNode.parentNode);
+            while (parentNode) {
+              snapOffsetX -= parentNode.position.x;
+              snapOffsetY -= parentNode.position.y;
+              parentNode = getNodes().find((node) => node.id === parentNode?.parentNode ?? '');
+            }
+            if (
+              helperLines.snapX &&
+              resizingChange.dimensions &&
+              movingChange.position &&
+              resizingNode.positionAbsolute &&
+              resizingNode.width
+            ) {
+              movingChange.position.x = helperLines.snapX + snapOffsetX;
+              resizingChange.dimensions.width =
+                resizingNode.width + resizingNode.positionAbsolute.x - helperLines.snapX;
+            }
+            if (
+              helperLines.snapY &&
+              resizingChange.dimensions &&
+              movingChange.position &&
+              resizingNode.positionAbsolute &&
+              resizingNode.height
+            ) {
+              movingChange.position.y = helperLines.snapY + snapOffsetY;
+              resizingChange.dimensions.height =
+                resizingNode.height + resizingNode.positionAbsolute.y - helperLines.snapY;
+            }
           }
         }
       }
-    }
-    return changes;
-  };
+      return changes;
+    },
+    [enabled]
+  );
 
-  const resetHelperLines = (changes: NodeChange[]): void => {
-    if (enabled && changes[0]?.type === 'reset') {
-      setState({ vertical: null, horizontal: null });
-    }
-  };
+  const resetHelperLines = useCallback(
```

Comment:
Beware using the reset event. This event is send by ReactFlow when using reactflowInstance.setNodes().
It might not always reflect the end of a move/reisze.
Here, the event is trigered by useLayoutOnBoundsChange setNodes().
So you might be looking at a similar condition than isBoundsChangeFinished there.

Reply:
done